File: SWD/Models/DonateDetail.cs
using System;
using System.Collections.Generic;

namespace SWD.Models
{
    public partial class DonateDetail
    {
        public int Id { get; set; }
        public int? CampaignId { get; set; }
        public int? UserId { get; set; }
        public double? Amount { get; set; }
        public DateTime? Date { get; set; }
        public string Description { get; set; }
        public int? GiftId { get; set; }

        public Campaign Campaign { get; set; }
        public GiftDetail Gift { get; set; }
        public User User { get; set; }
    }
}

File: lib/models/donate_detail/donate_detail.dart
class DonateDetail {
  int id;
  int campaignId;
  int userId;
  double amount;
  String date;
  int giftId;

  DonateDetail(
      {this.id,
      this.campaignId,
      this.userId,
      this.amount,
      this.date,
      this.giftId});

  DonateDetail.id({this.campaignId, this.userId, this.amount,
    this.date, this.giftId});


  factory DonateDetail.fromJson(Map<String, dynamic> json) {
    return DonateDetail(
        id: json['id'],
        campaignId: json['campaignId'],
        userId: json['userId'],
        amount: json['amount'],
        date: json['date'],
        giftId: json['giftId']);
  }

  Map toJson() => {
        'campaignId': campaignId,
        'userId': userId,
        'amount': amount,
        'date': date,
        'giftId': giftId
    };
}
